Mathlib/Analysis/Distribution/Distribution.lean

```diff
@@ -0,0 +1,168 @@
+/-
+Copyright (c) 2025 Anatole Dedecker. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Anatole Dedecker
+-/
+module
+
+public import Mathlib.Analysis.Distribution.TestFunction
+public import Mathlib.Analysis.LocallyConvex.StrongTopology
+
+/-!
+# Distributions
+
+Let `E` be a real **finite-dimensional normed space**, `Ω` an open subset of `E`,
+and `F` a real **locally convex topological vector space**.
+
+A **`F`-valued distributions on `Ω`** is a continuous `ℝ`-linear map `T : 𝓓(Ω, ℝ) →L[ℝ] F`,
+defined on the space `𝓓(Ω, ℝ)` of real-valued test functions, and taking values in `F`.
+In particular, if `𝕜` is `RCLike`, this is the usual notion of real or complex distribution.
+
+We denote the space of `F`-valued distributions on `Ω` by `𝓓'(Ω, F)`. Topologically,
+it is defined as `𝓓(Ω, ℝ) →L_c[ℝ] F`, meaning that we endow it with topology of uniform
+convergence on compact subsets of `𝓓(Ω, ℝ)`. If this choice of topology is surprising,
+see the implementation notes below.
+
+Right now, this file contains very few mathematical statements.
+The theory will be expanded very soon.
+
+## Main Declarations
+
+* `𝓓'^{n}(Ω, F) = Distribution Ω F n` is the space of `F`-valued distributions on `Ω` with
+  order at most `n`. See the implementation notes below for more information about the parameter
+  `n : ℕ∞`; in most cases you want to use the space `𝓓'(Ω, F) = Distribution Ω F ⊤`.
+* `Distribution.mapCLM`: any continuous linear map `A : F →L[ℝ] G` induces a continuous linear
+  map `𝓓'(Ω, F) →L[ℝ] 𝓓'(Ω, G)`. On locally integrable functions, this corresponds to applying `A`
+  pointwise.
+
+## Notation
+
+In the `Distributions` scope, we introduce the following notations:
+* `𝓓'^{n}(Ω, F)`: the space of `F`-valued distributions on the open set `Ω` with order at most
+  `n : ℕ∞`.
+* `𝓓'(Ω, F)`: the space of `F`-valued distributions on the open set `Ω`, i.e `𝓓'^{⊤}(Ω, F)`.
```

Comment:
Why do you use \top instead of \infty in this notation?

Reply:
I'm not sure to understand; but then on line 42 `n` should also be in `WithTop N\infty`? And also, \Top is for analytic functions, no?

Reply:
Sorry I was not very explicit.

The thing is, there is no analytic test function (other than zero), so the values of the order parameter (here and in [TestFunction](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Analysis/Distribution/TestFunction.html#TestFunction)) are `ℕ ∪ {∞}`, not `ℕ ∪ {∞, ω}`. This is why my `n` lives in `ℕ∞`, not `WithTop ℕ∞`.

Now, the `∞` and `ω` notations for smoothness exponents both denote elements of `WithTop ℕ∞`. More precisely, `∞ = (↑(⊤ : ℕ∞) : WithTop ℕ∞)` and `ω = (⊤ : WithTop ℕ∞)`. Since they are terms of type `WithTop ℕ∞`, I cannot use them for my parameter living in `ℕ∞`.

Is that clearer?

Reply:
Ah ok, the problem comes from the fact that ∞ is not a term of ℕ∞ (which is not terribly nice), but notation in this file inherits from the other one and so the only way to denote the top element in ℕ∞ is ⊤. Perhaps one small word to help someone reading this file but who ignores the library about Ck/C∞/Analytic=Cω functions?